lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,127 @@
+module Travis
+  module Build
+    class Script
+      class Elm < NodeJs
+        # Default NodeJS version to install
+        DEFAULT_NODE_VERSION = '6.10.0'
+
+        DEFAULTS = {
+          elm_version: '0.18.0',
+          elm_test_version: '0.18.7'
+        }
+
+        ELM_TEST_REQUIRED_NODE_VERSION = '4.0.0'
+
+        def export
+          super
+          sh.export 'TRAVIS_ELM_VERSION', elm_version, echo: false
+          sh.export 'TRAVIS_ELM_TEST_VERSION', elm_test_version, echo: false
+        end
+
+        def configure
+          super
+
+          config[:node_js] ||= DEFAULT_NODE_VERSION
+
+          sh.if '! -d sysconfcpus/bin' do
+            install_sysconfcpus
+          end
+        end
+
+        def announce
+          super
+          sh.cmd 'elm --version', echo: true
+          sh.cmd 'elm-test --version', echo: true
+        end
+
+        def setup
+          super
+
+          install_elm
+          install_elm_test
+        end
+
+        def script
+          sh.cmd 'elm-format --validate .'
```

Comment:
Does this command belong in `script`?

Reply:
Hm, I think it makes sense there if I understand the purpose of `script` correctly!

`elm-format --validate .` should fail the build if it returns a nonzero exit code (indicating someone has made a contribution that doesn't pass `elm-format` validation) and it shouldn't bother running tests until after that has been verified.

Reply:
At the moment, every command in `script` is executed (see https://github.com/travis-ci/travis-ci/issues/1066). So, the intention is to stop the build upon failure of this command, I suggest moving it to `install`.

Reply:
So I'd rather leave it in `script` because end users might want to override it as part of their test scripts. I changed it to use `&&` in 9e913d1f01d43fa138ce25e988971d1f9261ad64 - based on that thread, I figure this should do the trick, yeah?